harvard-huit/Harvard-Geospatial-Library
# Adding new credentials and RAILS_MASTER_KEY environment variable.

**Adding the new Rails master key so that the application can run as a Production Environment**
* * *

**JIRA Ticket**: none

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)
- https://medium.com/cedarcode/rails-5-2-credentials-9b3324851336
- https://secrethub.io/docs/guides/rails/

# What does this Pull Request do?
When switching the Rails environment to "production" it requires a master key that works with a credentials file. This pull request updates the key stored in credentials.yml.enc to match a new rails_master_key I generated for use in production mode. I also updated the env-example.txt to include the new RAILS_MASTER_KEY variable to store the key outside of the code repository. 

# How should this be tested?

A description of what steps someone could take to:
* To reproduce the original problem, just switch the RAILS_ENV in your .env file to "production" and restart the Docker container. You should see it fail in the logs telling you it is missing the master key.
* To fix the problem, you can add the RAILS_MASTER_KEY environment variable to the bottom of your .env file. I have already deployed the key itself to the .env files in the Dev and QA servers...so you can get the key from there to include in your .env
* After adding the RAILS_MASTER_KEY, restart the Docker container and the site should work normally as the Production environment.
* the "development" and "test" Rails environments do not require a master key. Setting the RAILS_ENV to either of those without having the RAILS_MASTER_KEY variable defined should also continue to work normally.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No 
- integration tests? No 

# Interested parties
@enriquediaz
